tangelo/toolboxes/molecular_computation/fno.py

```diff
@@ -0,0 +1,413 @@
+# Copyright 2023 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module containing datastructures for interfacing with the Frozen Natural
+Orbitals (FNOs), to automatically truncate the virtual orbital space.
+
+Reference: arXiv:2002.07901
+"""
+
+import warnings
+
+import numpy as np
+
+from tangelo.algorithms.classical import MP2Solver
```

Comment:
note to self: this does not seem to be the case here but I'm concerned that since algorithms depend on molecular computations we are here opening the gates to circular dependencies.

Reply:
I haven't added FNO to any `__init__.py` files, so I guess it shouldn't be imported when we import something from `molecular_computation`.

Reply:
I think I still don't fully understand the subtleties of imports, ill have to sit down and watch a video or two